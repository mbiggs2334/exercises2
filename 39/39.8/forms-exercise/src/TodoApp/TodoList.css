.TodoList-list {
    text-align: left;
}