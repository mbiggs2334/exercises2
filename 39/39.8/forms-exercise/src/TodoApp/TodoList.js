import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import './TodoList.css';

const TodoList = () => {

    const [todoList, setTodoList] = useState([]);

    const removeTodo = (todo) => {
        const idx = todoList.indexOf(todo);
        todoList.splice(idx, 1);
        setTodoList([...todoList]);
    };

    const addTodo = (newTodo) => {
        setTodoList([...todoList, newTodo]);
    };

    return (
        <>
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList">
                <ul className='TodoList-list'>
                    {todoList.map(val => 
                        <Todo key={uuid()} todo={val} removeTodo={removeTodo} />
                    )}
                </ul>
            </div>
        </>
        
    )
};

export default TodoList;