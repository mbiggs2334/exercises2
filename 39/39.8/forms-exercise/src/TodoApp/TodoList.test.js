import React from 'react';
import { render } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect';
import TodoList from './TodoList';

it('should render without crashing', () => {
    render(<TodoList />);
});

it('should match snapshot', () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});