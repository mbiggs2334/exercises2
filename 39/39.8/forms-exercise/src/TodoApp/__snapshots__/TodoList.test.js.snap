// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo: 
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="Enter a todo..."
      required=""
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
  <div
    class="TodoList"
  >
    <ul
      class="TodoList-list"
    />
  </div>
</DocumentFragment>
`;
